feat: Add height parameter to limit the box height

b-alidra opened the pull request:
This patch adds a `height` parameter to the `addTextbox` function.  
This height is used to limit the number of lines to write into the box.

b-alidra committed:
feat: Add height parameter to limit the box height

NikolaiMe commented:
Hello @b-alidra,
thank you for your contribution. This is basically a usefull feature. Unfortunately your current implementation breaks the existing implementation.

When I run my example code the expected result is the following:
![image](https://github.com/user-attachments/assets/93ab8001-2201-4977-8015-6c1e4dbef52a)

But when I ran the example with your code. I got the following result:
![image](https://github.com/user-attachments/assets/9f76dac1-5e82-4c94-b127-24716dbad54b)

The reason for this, is, that you added the `height` parameter in the addTextbox function in front of the `style` parameter.

`addTextbox(text,` doc, posX, posY, width, height, style = {})`

--> Therefore every user would need to change all the code which is currently using the style parameter to 

`addTextbox("text",` doc, 10, 15, 50, null, style = {})`

(the `null` needs to be added everywhere)

--> This means it is not downwards compatible and everyone who would update to the new package version would need to adapt the existing code to keep the package working. This is not good practice and can easily avoided by adding new optional parameters only to the very end of a function call.

Feel free to correct your implementation and provide an updated version. But please make sure that before you do the pull request, that you at least ran the two functions `testMe()` and `testMePartTwo()`, which are both part of "./examples/testPDFGenerator.js". Please compare then the resulting pdfs (test.pdf and testPartTwo.pdf) with the pdfs which can  be found in the gitHub repository. 

The best would be if you could add another testRoutine (e.g. testMePartThree()) which then also includes your new feature.

Short info: To really check your own code (and not the implementation which is right now on npm) you need to change the first line of testPDFGenerator.js to `const addTextbox = require("../js/pdfTextbox");`

NikolaiMe commented on README.md:
```diff
@@ -77,7 +77,7 @@ function testMe() {
   from the textbox.
   */
-  addTextbox(testTextArray, doc, 100, 100, 200, {
+  addTextbox(testTextArray, doc, 100, 100, 200, /*height*/null, {
```
Please remove the null to make it downward compatible

NikolaiMe commented on README.md:
````diff
@@ -103,14 +103,15 @@ const addTextbox = require("textbox-for-pdfkit");
 Syntax:
 ```javascript
-addTextbox(textArray, doc, posX, posY, width, defaultStyle);
+addTextbox(textArray, doc, posX, posY, width, height, defaultStyle);
````
change position of height and defaultStyle to make it downward compatible

NikolaiMe commented on README.md:
```diff
@@ -103,14 +103,15 @@ const addTextbox = require("textbox-for-pdfkit");
 - **posX:** The X-Position of the upper left corner of your Textbox (in PDF-points)
 - **posY:** The Y-Position of the upper left corner of your Textbox (in PDF-points)
 - **width:** The width of your Textbox (in PDF-points)
+- **height:** The maximum height of your Textbox (in PDF-points, optional)
```
Switch position of height and defaultStyle to make it downward compatible

NikolaiMe commented on js/pdfTextbox.js:
```diff
@@ -26,7 +26,7 @@ const defaultStyle = {
 // are defined)
 
-function addTextbox(text, doc, posX, posY, width, style = {}) {
+function addTextbox(text, doc, posX, posY, width, height, style = {}) {
```
Switch position of height and defaultStyle to make it downward compatible

NikolaiMe commented on js/pdfTextbox.js:
```diff
@@ -38,16 +38,21 @@ function addTextbox(text, doc, posX, posY, width, style = {}) {
   const baseline = style.baseline || "alphabetic";
 
-  drawTextLinesOnPDF(optimizedLines, width, posX, posY, textboxStyle, doc, baseline);
+  drawTextLinesOnPDF(optimizedLines, width, height, posX, posY, textboxStyle, doc, baseline);
```
Move the height parameter to the very end of the function

NikolaiMe commented on js/pdfTextbox.js:
```diff
@@ -38,16 +38,21 @@ function addTextbox(text, doc, posX, posY, width, style = {}) {
 // position of the PDF
 
-function drawTextLinesOnPDF(lines, width, posX, posY, defaultStyle, doc, baseline) {
+function drawTextLinesOnPDF(lines, width, height, posX, posY, defaultStyle, doc, baseline) {
```
Move the height parameter to the very end of the function

NikolaiMe reviewed (changes requested):
See my detailed comment above :)

b-alidra commented:
Hi @NikolaiMe and thank you for your review.

I've put the `height` parameter right after the `width` one because I think it makes sense but you're right about the downwards compatibility.
I changed my implementation and followed your remarks.
I've added a new test routine and the three are giving correct results.

Thanks again for your time.

NikolaiMe commented:
Wonderful! Thanks for your work and contribution! I will merge your branch into main now

jeremywillans commented:
@b-alidra, is there any way this could be expanded to vertically justify the text in the text box? 
If I have one line, i would like it to be in the middle, not at the top?

im thinking perhaps calculate total height of provided lines, and then use this to calculate the starting height.
